### PR TITLE
Support reveal_timeout in open channels API

### PR DESCRIPTION
**Pull request by andrevmatos**
Required for supporting this option in webUI.

### CODE REVIEW

**Comment by hackaugusto** [comment]
Why are we exposing the reveal timeout?

**Comment by LefterisJP** [comment]
Why not? It's part of the `open()` channel call, so we can expose it. If not given then we use the default, but the user (especially in the developer preview) should be able to adjust this value by doing something other than editing the code.

**Comment by hackaugusto** [comment]
> Why not? 

- Removing data is backwards incompatible, adding is not.
- Messing up with this value messes up with the safety of the software.
- The user may not understand that this value must be smaller than settle timeout

**Comment by LefterisJP** [comment]
> Messing up with this value messes up with the safety of the software.

This value can already be changed just from different interfaces (aka the python API or direct interaction with the blockchain.

Having a way to edit it from the default API (the REST API) is not a bad idea. Still we can remove it from the docs, so as to make it clear the user should not mess with this value unless he knows what he is doing.

**Comment by ulope** [comment]
> * The user may not understand that this value must be smaller than settle timeout

This constraint could be enforced in the Rest layer.

**Comment by andrevmatos** [comment]
I think the constraint is already enforced on the REST API (saw a exception being raised somewhere), and if we want to expose that on the UI, I can enforce that on the UI too. The point is, I don't know if it should be something that the user may set, it was a suggestion of @czepluch on giving the user this option.

File: raiden/tests/api/test_api.py [hunk 1]
```diff
@@ -434,10 +444,12 @@ def test_api_open_channel_invalid_input(
     partner_address = '0x61c808d82a3ac53231750dadc13c777b59310bd9'
     token_address = '0xea674fdde714fd979de3edf0f56aa9716b898ec8'
     settle_timeout = NETTINGCHANNEL_SETTLE_TIMEOUT_MIN - 1
+    reveal_timeout = 30
```
Comment (LefterisJP) [inline raiden/tests/api/test_api.py]: here and in other places where you use `30` shouldn't `DEFAULT_REVEAL_TIMEOUT` be used?

File: docs/rest_api.rst [hunk 2]
```diff
@@ -357,7 +357,8 @@ Example Request
         "partner_address": "0x61c808d82a3ac53231750dadc13c777b59310bd9",
```
Comment (LefterisJP) [inline docs/rest_api.rst]: If we do decide to add the `reveal_timeout` to the rest API then we most definitely should document here that it should **not be provided** in most circumstances. It's only an advanced user feature.

**Comment by LefterisJP** [comment]
@konradkonrad I would really appreciate your opinion on this PR.

As I stated in the chat: 

> I would like to be able to set it if for no other reason that I don't want to have to go and edit code in order to change it when testing or when creating a very specific channel topology.
> The documentation of the api call can explicitly state that this value is not to be messed with except by advanced users.

File: raiden/tests/api/test_api.py (hunk 1 above)
Comment (andrevmatos) [reply raiden/tests/api/test_api.py]: @LefterisJP most of those tests just use constants as testing parameters, but I can use `DEFAULT_REVEAL_TIMEOUT` too, of course.

File: docs/rest_api.rst (hunk 2 above)
Comment (andrevmatos) [reply docs/rest_api.rst]: Added observation in documentation discouraging messing with this option.

File: docs/rest_api.rst [hunk 3]
```diff
@@ -367,6 +368,10 @@ The ``balance`` field will signify the initial deposit you wish to make to the c
 The request to the endpoint should later return the fully created channel object
 from which we can find the address of the channel.
 
+**OBS:** DO NOT set ``reveal_timeout`` if you don't know what you're doing. This option
```
Comment (konradkonrad) [inline docs/rest_api.rst]: Is `**OBS:**` some special `.rst` tag? Otherwise I'd suggest using something more obvious, such as `**Caution:**`

**Comment by konradkonrad** [comment]
I would suggest to support it as an _undocumented_ feature at this point in time. Please add a test that ensures the API layer fills in the value, if missing from the request (at least a could not find such a test from cursory glance).

File: docs/rest_api.rst (hunk 3 above)
Comment (andrevmatos) [reply docs/rest_api.rst]: Done.

**Comment by LefterisJP** [comment]
@konradkonrad @andrevmatos I agree with that approach. Undocumented in the API, and not-existing in the Web-UI, but still possible to add it as an argument. That is the approach I would also like us to take.

**Comment by andrevmatos** [comment]
Done. @LefterisJP @konradkonrad addressed your comments.

**Comment by LefterisJP** [comment]
@andrevmatos Thank you. I will merge this since you performed the requested changes. We will have it as an undocumented argument of the API.

**Pull request merged**